compiler/src/iree/compiler/Codegen/Common/MaterializeEncodingIntoPackUnPack.cpp

```diff
@@ -946,6 +930,8 @@ void populateMaterializeEncodingIntoPackUnPackPatterns(
     MaterializeEncodingTypeConverter &typeConverter,
     MaterializeEncodingValueFn materializeEncodingValueFn) {
   MLIRContext *context = patterns.getContext();
+  // TODO(hanchung): Move the generic op pattern to ShapeIndependent category
+  // after we add the support for tile swizzling variants.
```

Comment:
Why is generic op going to ShapeIndependent? I thought it was not shape independent, since the indexing maps may change depending on the permutations/swizzles in MaterializeEncodingInfo.

Reply:
Yes, you're right. I still treat it as shape independent op because all the iterator types are parallel and the body is still the same. It is true that the indexing maps will change, but we will need to implement something similar for GPU data-tiling, mostly for dequant producers and perhaps elementwise consumers. We can revisit it later.